crates/ty_python_semantic/src/types.rs

```diff
@@ -4038,6 +4033,23 @@ impl<'db> Type<'db> {
             }
 
             (Type::GenericAlias(_), _) | (_, Type::GenericAlias(_)) => ConstraintSet::from(true),
+
+            (Type::TypedDict(self_typeddict), Type::TypedDict(other_typeddict)) => {
+                disjointness_visitor.visit((self, other), || {
+                    self_typeddict.is_disjoint_from_impl(
+                        db,
+                        other_typeddict,
+                        inferable,
+                        disjointness_visitor,
+                        relation_visitor,
+                    )
+                })
+            }
+
+            (Type::TypedDict(_), _) | (_, Type::TypedDict(_)) => {
+                // TODO: Implement disjointness for TypedDict with other types.
```

Comment:
Does this `TODO` need to be here? Or could instead have a comment that says something like "all the relevant cases (like `Mapping`) have been handled above"? (Edit: I should probably add a test that that has been handled...)

Reply:
I added a "Disjointness with other types" section to the mdtests.